### DISCUSSION TITLE
Api github java

### DISCUSSION THREAD

**Discussion by ArtMiProg** [community/community#64862] [Apps, API and Webhooks]

**Comment by arafajs** [comment]
Reasons for this:

- The user does not exist on GitHub.
- The user has not made any public repositories.
- You are not authenticated with GitHub, and the user has their repositories set to private.

To fix this, you can try the following:

1. Make sure that the user you are trying to list the repositories for exists on GitHub. You can do this by going to their GitHub profile page.
2. If the user exists, make sure that they have made at least one public repository. You can do this by checking their repository list.
3. If the user has made public repositories, make sure that you are authenticated with GitHub. You can do this by creating a GitHub account and logging in.
4. If you are authenticated with GitHub, make sure that the user has not set their repositories to private. You can check this by looking at the visibility of their repositories.

**Reply by arafajs** [reply]
In that case, the problem is most likely with your Spring REST app

- Make sure that you are calling the correct method in your Spring REST app. The method for listing all of a user's repositories is `getRepositories()`.
- Make sure that you are passing the correct username to the method.
- Make sure that you are handling the 404 error correctly in your Spring REST app. You can do this by creating a custom exception handler.

**Reply by arafajs** [reply]
You may not be calling the endpoint correctly from the client side. Double check that the URL you are trying in the browser/Postman matches the endpoint URL pattern in your controller.
